src/active_query.rs

```diff
@@ -116,30 +143,99 @@ impl ActiveQuery {
         self.input_outputs.contains(&QueryEdge::Output(key))
     }
 
-    pub(crate) fn into_revisions(self) -> QueryRevisions {
-        let edges = QueryEdges::new(self.input_outputs);
-        let origin = if self.untracked_read {
+    fn take_revisions(&mut self) -> QueryRevisions {
+        let &mut Self {
+            database_key_index: _,
+            disambiguator_map: _,
+            durability,
+            changed_at,
+            untracked_read,
+            accumulated_inputs,
+            ref mut cycle_heads,
+            ref mut input_outputs,
+            ref mut tracked_struct_ids,
+            ref mut accumulated,
+        } = self;
+
+        let edges = QueryEdges::new(input_outputs.drain(..));
```

Comment:
Should we use `mem::take` here too (and change `QueryEdges::new` to take a `Box<[QueryEdge]>`) to avoid the extra allocation?

Reply:
`input_outputs` is an `IndexSet` so that won't work